src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1405,6 +1515,71 @@ private void WritePackageDependencies()
                 }
             }
 
+            private void WritePackageDependenciesBetweenPackages()
+            {
+                foreach(var target in _lockFile.Targets)
+                {
+                    GetPackageDependencies(target);
+                }
+
+                void GetPackageDependencies(LockFileTarget target)
+                {
+                    var resolvedPackageVersions = target.Libraries
+                    .ToDictionary(pkg => pkg.Name, pkg => pkg.Version.ToNormalizedString(), StringComparer.OrdinalIgnoreCase);
+
+                    string frameworkAlias = _targetNameToAliasMap[target.Name];
+
+                    var transitiveProjectRefs = new HashSet<string>(
+                        target.Libraries
+                            .Where(lib => lib.IsTransitiveProjectReference(_lockFile, ref _projectFileDependencies, frameworkAlias))
+                            .Select(pkg => pkg.Name),
+                        StringComparer.OrdinalIgnoreCase);
+
+                    foreach (var package in target.Libraries)
+                    {
+                        string packageId = $"{package.Name}/{package.Version.ToNormalizedString()}";
```

Comment:
This allocation can be moved within the `if` block beneath, so we only allocate when needed.

Reply:
Fixed in https://github.com/dotnet/sdk/pull/28263/commits/5e8fd8b95fcf519d363ce9325ef27a1999f6fb2b